src/bin/console/console.cpp

```diff
@@ -281,15 +270,9 @@ int Console::_eval_sql(const std::string& sql) {
   out(std::to_string(row_count) + " rows total (PARSE: " + std::to_string(parse_elapsed_ms) + " ms, COMPILE: " +
       std::to_string(plan_elapsed_ms) + " ms, EXECUTE: " + std::to_string(execution_elapsed_ms) + " ms (wall time))\n");
 
-  if (transaction_context->aborted()) {
-    out("An operator failed and the transaction has been rolled back.\n");
-    _explicitly_created_transaction_context = nullptr;
-    return ReturnCode::Error;
-  }
-
-  // Commit
-  if (auto_commit) {
-    transaction_context->commit();
+  // Default auto-commit if no context is set explicitly
+  if (!_explicitly_created_transaction_context) {
+    _sql_pipeline->transaction_context()->commit();
```

Comment:
This should be part of the pipeline. If the pipeline creates its own TXContext, it should also do the commit internally.